From 93b5f11641fa22d4518f251d6e3db99219864677 Mon Sep 17 00:00:00 2001
From: Brandon Wright <bearoso@gmail.com>
Date: Sat, 6 May 2017 18:43:36 -0500
Subject: [PATCH] Fix environment variable name.

---
 gtk/src/gtk_config.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gtk/src/gtk_config.cpp b/gtk/src/gtk_config.cpp
index b957d1f4..b4db218c 100644
--- a/gtk/src/gtk_config.cpp
+++ b/gtk/src/gtk_config.cpp
@@ -41,7 +41,7 @@ get_config_dir (void)
 
     /* Find config directory */
     home_dir = getenv ("HOME");
-    xdg_config_dir = getenv ("XDG_CONFIG_DIR");
+    xdg_config_dir = getenv ("XDG_CONFIG_HOME");
 
     if (!home_dir && !xdg_config_dir)
     {
